app/src/hooks/useMarketMakerData.tsx

```diff
@@ -0,0 +1,160 @@
+  const [marketMakerUserFunding, setMarketMakerUserFunding] = useState<BigNumber>(new BigNumber(0))
+  const [marketMakerFundingPercentage, setMarketMakerFundingPercentage] = useState<number>(0)
+  const [collateral, setCollateral] = useState<Maybe<Token>>(null)
+
+  useEffect(() => {
+    const fetchMarketMakerData = async ({ enableStatus }: { enableStatus: boolean }) => {
+      try {
+        enableStatus && setStatus(Status.Loading)
+        const provider = context.library
+        const user = await provider.getSigner().getAddress()
+
+        const marketMaker = new MarketMakerService(marketMakerAddress, conditionalTokens, provider)
+
+        const conditionId = await marketMaker.getConditionId()
+        const isConditionResolved = await conditionalTokens.isConditionResolved(conditionId)
+
+        const winnerOutcomeData = isConditionResolved
+          ? await conditionalTokens.getWinnerOutcome(conditionId)
+          : null
+        setWinnerOutcome(winnerOutcomeData)
+
+        const [
+          userShares,
+          marketMakerShares,
+          actualPrices,
+          marketMakerFund,
+          marketMakerUserFund,
+          collateralAddress,
+        ] = await Promise.all([
+          marketMaker.getBalanceInformation(user),
+          marketMaker.getBalanceInformation(marketMakerAddress),
+          // TODO: calculate actual prices
+          // marketMaker.getActualPrice(),
+          { actualPriceForYes: 0.5, actualPriceForNo: 0.5 },
+          marketMaker.getTotalSupply(),
+          marketMaker.balanceOf(user),
+          marketMaker.getCollateralToken(),
+        ])
+
+        const token = getTokenFromAddress(context.networkId, collateralAddress)
+        setCollateral(token)
+
+        const probabilityForYes = actualPrices.actualPriceForYes * 100
+        const probabilityForNo = actualPrices.actualPriceForNo * 100
+
+        const balanceShares = [
+          {
+            outcomeName: OutcomeSlot.Yes,
+            probability: Math.round((probabilityForYes / 100) * 100),
+            currentPrice: actualPrices.actualPriceForYes,
+            shares: userShares.balanceOfForYes,
+            holdings: marketMakerShares.balanceOfForYes,
+            winningOutcome: winnerOutcome === WinnerOutcome.Yes,
+          },
+          {
+            outcomeName: OutcomeSlot.No,
+            probability: Math.round((probabilityForNo / 100) * 100),
+            currentPrice: actualPrices.actualPriceForNo,
+            shares: userShares.balanceOfForNo,
+            holdings: marketMakerShares.balanceOfForNo,
+            winningOutcome: winnerOutcome === WinnerOutcome.No,
+          },
+        ]
+
+        const totalMarketMakerShares = marketMakerShares.balanceOfForNo.add(
+          marketMakerShares.balanceOfForYes,
+        )
+
+        const totalUserShares = userShares.balanceOfForNo.add(userShares.balanceOfForYes)
+        const totalUserSharesNumber = +ethers.utils.formatUnits(totalUserShares, 18)
```

Comment:
This should use the collateral token's decimals.

Reply:
Done